R/runCCA.R

```diff
@@ -157,311 +155,394 @@
-        cca <- vegan::cca(X = x, scale = scale, na.action = na.action, ...)
-        X <- cca$CA
-    }
-    # Create the matrix to be returned
-    ans <- X[[scores]]
-    attr(ans, "rotation") <- X$v
-    attr(ans, "eigen") <- X$eig
-    attr(ans, "cca") <- cca
-    ans
+#' @export
+#' @rdname runCCA
+#' @aliases getCCA
+calculateCCA <- function(x, ...){
+    getCCA(x, ...)
 }
 
 #' @export
 #' @rdname runCCA
-setMethod("getCCA", "ANY",
-      function(x, ...){
-          .calculate_cca(x, ...)
-      })
+#' @aliases addCCA
+runCCA <- function(x, ...){
+    addCCA(x, ...)
+}
 
-#' @importFrom stats terms
-#' @importFrom SummarizedExperiment colData
-.get_variables_from_data_and_formula <- function(x, formula){
-    if(missing(formula)){
-        return(NULL)
+#' @export
+#' @rdname runCCA
+setMethod("getCCA", "ANY", function(x, formula, data, ...){
+    if( !is.matrix(x) ){
+        stop("'x' must be matrix.", call. = FALSE)
     }
-    terms <- rownames(attr(terms(formula),"factors"))
-    terms <- terms[terms != as.character(formula)[2L]]
-    terms <- .remove_special_functions_from_terms(terms)
-    if(!all(terms %in% colnames(colData(x)))){
-        stop("All variables on the right hand side of 'formula' must be ",
-             "present in colData(x).", call. = FALSE)
+    if( !is(formula, "formula") ){
+        stop("'formula' must be formula or NULL.", call. = FALSE)
     }
-    colData(x)[,terms,drop=FALSE]
-}
-
-.get_formula_from_data_and_variables <- function(x, variables){
-    # Check that variables specify columns from colData
-    if( all(!is.character(variables)) ){
-        stop("'variables' should be a character specifying variables from ",
-             "colData(x).", 
-             call. = FALSE)
+    if( !(is.data.frame(data) || is.matrix(data) || is(data, "DFrame")) ){
+        stop("'data' must be data.frame or coarcible to one.", call. = FALSE)
     }
-    if(!all(variables %in% colnames(colData(x)))){
-        stop("All variables must be ",
-             "present in colData(x).", call. = FALSE)
+    if( ncol(x) != nrow(data) ){
+        stop("Number of columns in 'x' should match with number of rows in ",
+            "'data'.", call. = FALSE)
     }
-    # Create a formula based on variables
-    formula <- as.formula(paste0("x ~ ", 
-                                 paste(variables, collapse = " + ")))
-    formula
-}
+    #
+    res <- .calculate_rda(
+        x, formula = formula, data = data, ord.method = "CCA", ...)
+    return(res)
+    })
 
 #' @export
 #' @rdname runCCA
 setMethod("getCCA", "SummarizedExperiment",
-    function(x, formula, variables, test.signif = TRUE,
-             assay.type = assay_name, assay_name = exprs_values, exprs_values = "counts",
-             scores = "wa", ...)
-    {
-        # Check assay.type and get assay
+    function(
+        x, formula = NULL, col.var = variables, variables = NULL,
+        test.signif = TRUE, assay.type = assay_name, assay_name = exprs_values,
```

Comment:
vairables are renamed to col.var. That is because they are column names from colData

Reply:
We also use col.var in other functions